src/main/java/picard/sam/SamErrorMetric/ReadBaseStratification.java

```diff
@@ -476,6 +481,74 @@ public String getSuffix() {
         }
     }
 
+    /**
+     * Stratifies according to the number of matching cigar operators (from CIGAR string) that the read has.
+     */
+    public static class CigarOperatorsInReadStratifier extends RecordStratifier<Integer> {
+
+        private CigarOperator operator;
+
+        public CigarOperatorsInReadStratifier(final CigarOperator op) {
+            operator = op;
+        }
+
+        @Override
+        public Integer stratify(final SAMRecord samRecord) {
+            try {
+                return samRecord.getCigar().getCigarElements().stream()
+                        .filter(ce -> ce.getOperator().equals(operator))
+                        .mapToInt(CigarElement::getLength)
+                        .sum();
+            } catch (final Exception ex) {
+                return null;
+            }
+        }
+
+        @Override
+        public String getSuffix() {
+            return "cigar_elements_" + operator.name() + "_in_read";
+        }
+    }
+
+    /**
+     * Stratifies according to the number of indel bases (from CIGAR string) that the read has.
+     */
+    public static class IndelsInReadStratifier extends RecordStratifier<Integer> {
```

Comment:
Should this be implemented as a sum of two `CigarOperatorsInReadStratifier` (`I` and `D`)? probably not as fast, but less code repetition?

Reply:
Good idea. I moved the actual counting to a private static method so that we don't need to instantiate two Stratifier objects to get the result. (That probably made it more code overall, but less repetition.)